### PR TITLE
chore: explore compiler class

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #4233

### Description:

Changes:
- move metric SQL logic to each warehouse client
- convert explore compiler functions into a class ( this is going to be hard to review since the entire file shows as changed, but the unit tests and mocks are pretty much unchanged )

### CODE REVIEW

**Comment by rephus** [comment]
Is this PR also doing `We should also move the quoteChart, etc to the warehouse client and make it available via a method.` ? or that will be done a separate PR

File: packages/warehouses/src/warehouseClients/BigqueryWarehouseClient.ts [hunk 1]
```diff
@@ -256,4 +259,15 @@ export class BigqueryWarehouseClient implements WarehouseClient {
     getEscapeStringQuoteChar() {
         return '\\';
     }
+
+    getMetricSql(sql: string, metric: Metric) {
+        switch (metric.type) {
+            case MetricType.PERCENTILE:
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/BigqueryWarehouseClient.ts:265]: so all metric types we have so far are the same for all warehouses (getDefaultMetricSql) , but this PERCENTILE is different per warehouse, right ?

**Review by rephus — APPROVED** [review]
I'm not sure how to test this, the refactoring makes sense. I guess if e2e pass then we can merge

File: packages/warehouses/src/warehouseClients/BigqueryWarehouseClient.ts (hunk 1 above)
Comment (ZeRego) [reply packages/warehouses/src/warehouseClients/BigqueryWarehouseClient.ts:265]: correct

**Comment by ZeRego** [comment]
> We should also move the quoteChart, etc to the warehouse client and make it available via a method.

@rephus This work was merged in a previous PR.

**Pull request merged**